File: src/main/java/org/example/common/tokens/Keywords.java
package org.example.common.tokens;

public enum Keywords {
    VAR, LET, CONST, CONSOLE, LOG
}

// Por el momento solo hay let.
// DUDA: ¿incluir string y number como keywords?
// ¿Si no aislar let a "DeclaratorToken"? Problema con límite de extensibilidad?
// If so, ¿qué pasa con Keywords...?

File: src/main/kotlin/org.example/common/tokens/Keywords.kt
package org.example.common.tokens

enum class Keywords() {
    VAR, LET, CONST, CONSOLE, LOG;

    companion object {
        fun fromString(word: String): Keywords =
            Keywords.entries.first { it.name.equals(word, ignoreCase = true) }
    }
}

// Por el momento solo hay LET.
// DUDA: ¿incluir 'string' y 'number' como keywords?
// ¿Si no, aislar LET a "DeclaratorToken"? Problema con límite de extensibilidad?
// Si es así, ¿qué pasa con Keywords...?